### PR TITLE
Oh 702

### PR DESCRIPTION
**Pull request by mikaelbarlund**
adding types
changed a:s to Links

### CODE REVIEW

File: organisaatio-ui/src/components/Sivut/LomakeSivu/LomakeSivu.tsx [hunk 1]
```diff
@@ -149,26 +170,56 @@ const LomakeSivu = (props: any) => {
...
-                        </a>
+                        <Link to={`${o.oid}`}>{o.nimi[language] || o.nimi['fi'] || o.nimi['sv'] || o.nimi['en']}</Link>
                     </div>,
                     organisaatioNimiPolku.length - 1 !== index && <div> &gt; </div>,
                 ])}
             </div>
             <div className={styles.ValiContainer}>
                 <div className={styles.ValiOtsikko}>
-                    <h3>{organisaatio.tyypit[0]}</h3>
+                    <h3>{organisaatio.tyypit ? organisaatio.tyypit[0] : 'N/A'}</h3>
```
Comment (jollem) [inline organisaatio-ui/src/components/Sivut/LomakeSivu/LomakeSivu.tsx:222]: hardcoded 'N/A'?

File: organisaatio-ui/src/components/Sivut/LomakeSivu/UusiToimija/PerustietoLomake/PerustietoLomake.tsx [hunk 2]
```diff
@@ -46,14 +53,15 @@ export default function PerustietoLomake(props: OrganisaatioProps) {
...
+    const kaikkiKunnat = kuntaKoodisto.koodit().map((k) => ({
         value: k.uri,
-        label: k.nimi[language] || k.nimi['fi'] || k.nimi['sv'] || k.nimi['en'],
+        label: k.nimi[language] || k.nimi['fi'] || k.nimi['sv'] || k.nimi['en'] || '',
     }));
     const [nimi, setNimi] = useState(tyhjaNimi);
 
     const handleNimiTallennus = () => {
         const nimet = { nimi: Object.assign({}, nimi), alkuPvm: new Date().toISOString().split('T')[0] };
+        console.log(nimet);
```
Comment (jollem) [inline organisaatio-ui/src/components/Sivut/LomakeSivu/UusiToimija/PerustietoLomake/PerustietoLomake.tsx:64]: maybe redundant to output these?

File: organisaatio-ui/src/components/Sivut/LomakeSivu/UusiToimija/PerustietoLomake/PerustietoLomake.tsx [hunk 3]
```diff
@@ -152,13 +162,13 @@ export default function PerustietoLomake(props: OrganisaatioProps) {
...
-                    <label>{i18n.translate('PERUSTIETO_MUUT_KUNNAT')}</label>
+                    <label>{i18n.translate('PERUSTIETO_MUUT_KUNNATs')}</label>
                     <Select
                         isMulti
                         value={
-                            organisaatio.muutKotipaikatUris && organisaatio.muutKotipaikatUris.length
-                                ? kaikkiKunnat.filter(
-                                      (kk) => !!organisaatio.muutKotipaikatUris.find((mku: string) => mku === kk.value)
+                            organisaatio.muutKotipaikatUris
+                                ? organisaatio.muutKotipaikatUris.map(
```
Comment (jollem) [inline organisaatio-ui/src/components/Sivut/LomakeSivu/UusiToimija/PerustietoLomake/PerustietoLomake.tsx:170]: dunno but could the ternary be replaced with `(organisaatio.muutKotipaikatUris || [])`?

File: organisaatio-ui/src/components/Sivut/LomakeSivu/UusiToimija/PerustietoLomake/PerustietoLomake.tsx [hunk 4]
```diff
@@ -202,18 +212,18 @@ export default function PerustietoLomake(props: OrganisaatioProps) {
...
+                            .find((mv) => {
                                 return organisaatio.kieletUris.find(
                                     (kU: string) => kU.slice(0, kU.length - 2) === mv.value
                                 );
                             })}
-                        options={opetuskielet.map((mv: any) => ({
+                        options={opetuskielet.map((mv) => ({
                             value: `${mv.uri}#${mv.versio}`,
-                            label: mv.nimi[language] || mv.nimi['fi'] || mv.nimi['sv'] || mv.nimi['en'],
+                            label: mv.nimi[language] || mv.nimi['fi'] || mv.nimi['sv'] || mv.nimi['en'] || '',
```
Comment (jollem) [inline organisaatio-ui/src/components/Sivut/LomakeSivu/UusiToimija/PerustietoLomake/PerustietoLomake.tsx:226]: this seems to repeat a lot and has "logic". consider dedicated function?

File: organisaatio-ui/src/components/Sivut/Tyypit/Muokkaus/LisatietotyypinMuokkaus.tsx [hunk 5]
```diff
@@ -20,15 +20,15 @@ const KAIKKIVALITTU = '1',
...
+    match: { params: { nimi: string } };
 };
 
 export default function LisatietotyypinMuokkaus(props: Props) {
     const { i18n, language } = useContext(LanguageContext);
     const [lisatietotyyppi, setLisatietotyyppi] = useState<undefined | any>();
     const [{ data: oppilaitosTyypit, loading: oppilaitosTyypitLoading, error: oppilaitosTyypitError }] = useAxios<
         Koodi[]
-    >(`/organisaatio/koodisto/OPPILAITOSTYYPPI/koodi?noCache=1595328878067&onlyValidKoodis=true`);
+    >(`/organisaatio/koodisto/OPPILAITOSTYYPPI/koodi?noCache=1595328878067&onlyValidKoodis=true`); //TODO the random parameter should probably be random
```
Comment (jollem) [inline organisaatio-ui/src/components/Sivut/Tyypit/Muokkaus/LisatietotyypinMuokkaus.tsx:31]: yes it should. timestamp would probably suffice. do fix

**Review by jollem — APPROVED** [review]
Good stuff 👌

File: organisaatio-ui/src/components/Sivut/LomakeSivu/LomakeSivu.tsx (hunk 1 above)
Comment (mikaelbarlund) [reply organisaatio-ui/src/components/Sivut/LomakeSivu/LomakeSivu.tsx:222]: true, should "never" end up there though

File: organisaatio-ui/src/components/Sivut/LomakeSivu/UusiToimija/PerustietoLomake/PerustietoLomake.tsx (hunk 2 above)
Comment (mikaelbarlund) [reply organisaatio-ui/src/components/Sivut/LomakeSivu/UusiToimija/PerustietoLomake/PerustietoLomake.tsx:64]: I think we need a separate issue to remove all development log outputs

File: organisaatio-ui/src/components/Sivut/LomakeSivu/UusiToimija/PerustietoLomake/PerustietoLomake.tsx (hunk 3 above)
Comment (mikaelbarlund) [reply organisaatio-ui/src/components/Sivut/LomakeSivu/UusiToimija/PerustietoLomake/PerustietoLomake.tsx:169]: yes, I actually like this

**Review by mkamka — APPROVED** [review]

**Pull request merged**